script/deploy/safe/deploy-safe.ts

```diff
@@ -0,0 +1,696 @@
+      )
+    }
+
+    // Get owner addresses from global.json
+    const ownerAddresses = globalConfig.safeOwners as string[]
+    if (!ownerAddresses || ownerAddresses.length === 0) {
+      throw new Error('No Safe owners found in config/global.json')
+    }
+
+    consola.info('-'.repeat(80))
+    consola.info(`🔐 Deploying new Safe Wallet on ${network.toUpperCase()}`)
+    consola.info('-'.repeat(80))
+
+    // Initialize Viem clients
+    const chain = getViemChainForNetworkName(network)
+    const customRpcUrl = rpcUrl || chain.rpcUrls.default.http[0]
+
+    const account = privateKeyToAccount(`0x${privateKey}` as Hex)
+
+    const publicClient = createPublicClient({
+      chain,
+      transport: http(customRpcUrl),
+    })
+
+    const walletClient = createWalletClient({
+      account,
+      chain,
+      transport: http(customRpcUrl),
+    })
+
+    consola.info(
+      `Network: \u001b[33m${network} (Chain ID: ${chain.id})\u001b[0m`
+    )
+    consola.info(`Deployer address: \u001b[32m${account.address}\u001b[0m`)
+    consola.info(`RPC URL: ${customRpcUrl}`)
+    consola.info('-'.repeat(80))
+
+    // Deploy the Safe
+    const safeAddress = await deployNewSafe({
+      publicClient,
+      walletClient,
+      chain,
+      ownerAddresses: ownerAddresses as Address[],
+      threshold: 3,
+    })
+
+    if (!safeAddress) {
+      throw new Error('Safe deployment failed')
+    }
+
+    consola.success(
+      `✅ Safe successfully deployed at \u001b[32m${safeAddress}\u001b[0m`
+    )
+
+    // Update networks.json if enabled
+    if (updateConfig) {
+      // Determine Safe Web URL based on chain
+      let safeWebUrl = ''
+      if (chain.id === 1) {
+        // Ethereum Mainnet
+        safeWebUrl = `https://app.safe.global/home?safe=eth:${safeAddress}`
+      } else {
+        // Other chains, use network-specific format
+        const networkKey = network.toLowerCase()
+        let chainShortName = networkKey
+
+        // Map to Safe's standard chain names if needed
+        const safeChainNameMap: Record<string, string> = {
+          arbitrum: 'arb1',
+          optimism: 'oeth',
+          polygon: 'matic',
+          bsc: 'bnb',
+          // Add more mappings as needed
+        }
+
+        if (safeChainNameMap[networkKey]) {
+          chainShortName = safeChainNameMap[networkKey]
+        }
+
+        safeWebUrl = `https://app.safe.global/home?safe=${chainShortName}:${safeAddress}`
+      }
+
+      // Update configurations
+      const updatedNetworksConfig = { ...networksConfig }
+      updatedNetworksConfig[network.toLowerCase()].safeAddress = safeAddress
+      updatedNetworksConfig[network.toLowerCase()].safeWebUrl = safeWebUrl
+
+      // Create backup of networks.json
+      const backupPath = `${networksConfigPath}.backup-${Date.now()}`
+      fs.copyFileSync(networksConfigPath, backupPath)
+      consola.info(`Created backup of networks.json at ${backupPath}`)
```

Comment:
Would it make sense to delete that backup after the update?

Reply:
Removed. Not really needed if we have a git history.